### PR TITLE
Account for podman version < 2.0.0

### PR DESCRIPTION
**Pull request by dtantsur**
The default for CentOS 8 is 1.6.4 which does not support units
in `podman wait -i`. Add a version check to support both.

Follow-up to commit 42af6102b4901a1818e7e71607838b1d9a8c36a9.

### CODE REVIEW

File: 04_setup_ironic.sh [hunk 1]
```diff
@@ -12,6 +12,8 @@ source validation.sh
 
 early_deploy_validation
 
+PODMAN_VERSION=$(podman version -f json | jq -r .Version)
+
```
Comment (hardys) [inline 04_setup_ironic.sh:16]: Ugh, the version output also isn't compatible, so this needs to be something like `podman version -f json | jq -r '.Version,.Client.Version|strings'`

```
$ podman version -f json | jq .
{
  "Client": {
    "APIVersion": 1,
    "Version": "2.0.5",
    "GoVersion": "go1.14.7",
    "GitCommit": "",
    "BuiltTime": "Wed Sep 23 17:18:02 2020",
    "Built": 1600877882,
    "OsArch": "linux/amd64"
  }
}
```

Comment (hardys) [reply 04_setup_ironic.sh:16]: Also probably best to add sudo for consistency (although I think rootless podman somewhat works now, this version command actually fails on centos7 without sudo...)

Comment (dtantsur) [reply 04_setup_ironic.sh:17]: Wow. Thank you Steve, updating.

File: 04_setup_ironic.sh [hunk 2]
```diff
@@ -12,6 +12,9 @@ source validation.sh
 
 early_deploy_validation
 
+# Account for differences in 1.* and 2.* version reporting.
+PODMAN_VERSION=$(sudo podman version -f json | jq -r '.Version,.Client.Version|strings')
```
Comment (andfasano) [inline 04_setup_ironic.sh:16]: Shorter version could be ```sudo podman version --format "{{.Version}}"```

File: 04_setup_ironic.sh [hunk 3]
```diff
@@ -160,7 +163,12 @@ then
   sudo podman run -d --net host --privileged --name ipa-downloader --pod ironic-pod \
      -v $IRONIC_DATA_DIR:/shared ${IRONIC_IPA_DOWNLOADER_LOCAL_IMAGE} /usr/local/bin/get-resource.sh
 
-  sudo podman wait -i 1000ms ipa-downloader
+  # Units have been introduced in 2.x
+  if printf '2.0.0\n%s\n' "$PODMAN_VERSION" | sort -V -C; then
+      sudo podman wait -i 1000ms ipa-downloader
+  else
+      sudo podman wait -i 1000 ipa-downloader
+  fi
```
Comment (andfasano) [inline 04_setup_ironic.sh:171]: Question: do we really need to maintain the backward compatibility to a previous (major) version? Wouldn't be simpler to bump the podman min version required by dev-scripts?

Comment (stbenjam) [reply 04_setup_ironic.sh:171]: No. CentOS doesn't have a podman that understands units, and RHEL 8.3 requires them.

File: 04_setup_ironic.sh (hunk 2 above)
Comment (stbenjam) [reply 04_setup_ironic.sh:16]: The format argument to podman doesn't work on some versions.

Comment (dtantsur) [reply 04_setup_ironic.sh:16]: And it's .Client.Version on newer podmans..

File: 04_setup_ironic.sh (hunk 3 above)
Comment (andfasano) [reply 04_setup_ironic.sh:171]: Ok got it just for this command , not in general

File: 04_setup_ironic.sh (hunk 2 above)
Comment (andfasano) [reply 04_setup_ironic.sh:16]: There's also a .Server.Version but it seems both are supported

**Pull request merged**